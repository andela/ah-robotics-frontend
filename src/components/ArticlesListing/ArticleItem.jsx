import React from 'react';
import { Item, Label } from 'semantic-ui-react';
import renderHTML from 'react-render-html';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import TruncateMarKup from 'react-truncate-markup';

const ArticleItem = ({ article }) => ({ article }
  && (
  <Item>
    <Item.Image src={(article.image) ? article.image : 'https://source.unsplash.com/random/400x300'} />
    <Item.Content>
      <Item.Header as="a">
        <NavLink to={`article/${article.slug}`}>{article.title}</NavLink>
      </Item.Header>
      <TruncateMarKup lines={12}>
        <p>{renderHTML(article.body)}</p>
      </TruncateMarKup>
      <Item.Extra>
        <Label className="author-name">{`By ${article.author.username}`}</Label>
        <Label content={`Read time: ${article.read_time}`} icon="time" />
      </Item.Extra>
    </Item.Content>
  </Item>
)
);
ArticleItem.propTypes = {
  article: PropTypes.shape({}).isRequired,
};
export default ArticleItem;
