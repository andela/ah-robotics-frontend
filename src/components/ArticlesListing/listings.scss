.lowerSection{
  margin-top: 50px !important;
}
.ui.items>.item>.image+.content{
  padding: 1.5em;
  min-height:300px ;
  min-width: 34em;
}
.ui.divided.items>.item {
  border-top: none;
}
.ui.items>.item>.image>img {
  padding-top: 1.5em;
  padding-bottom: 0.2em;
  max-height: 300px;
  height: auto;
  display: block;
  margin-left: auto;
  margin-right: auto;
  width: 100%;
}
.ui.items > .item > .image + .content {
  margin-left: 0.5em;
}
.ui.items>.item>.image:not(.ui){
  width: 400px !important;
  height: auto;
  max-height: 300px;
}
.author-name{
 text-transform: capitalize;
}
.ui.basic.segment {
  margin-left: 4.5em;
}
.content .header > a{
  text-transform:capitalize ;
}
