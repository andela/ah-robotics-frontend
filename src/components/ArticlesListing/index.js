import React from 'react';
import {
  Grid, Segment, Item,
} from 'semantic-ui-react';
import './listings.scss';
import PropTypes from 'prop-types';
import ArticleItem from './ArticleItem';

const ArticlesListing = ({ articles, isFetching }) => {
  const allArticles = articles.articles;
  return (
    <Segment basic loading={isFetching}>
      <Grid>
        <Grid.Column width={12}>
          <Item.Group divided>
            {allArticles && allArticles.map(article => (
              (<ArticleItem article={article} />)
            ))}
          </Item.Group>
        </Grid.Column>
        <Grid.Column width={4}>
        </Grid.Column>
      </Grid>
    </Segment>
  );
};

ArticlesListing.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  isFetching: PropTypes.bool.isRequired,
};
export default ArticlesListing;
