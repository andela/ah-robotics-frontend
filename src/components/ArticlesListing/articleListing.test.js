import React from 'react';
import ArticlesListing from '.';
import ArticleItem from './ArticleItem';

const article = {
  title: 'hey',
  description: 'hey',
  body: 'this is ...',
  read_time: '034',
  author: {
    username: 'joh doe',
  },

};
const isFetching = false;
const articles = [{}];

it('should render article listing component', () => {
  const wrapper = shallow(<ArticlesListing isFetching={isFetching} articles={articles} />);
  expect(wrapper.length).toBe(1);
});
it('article listing component should match snapshot', () => {
  const wrapper = shallow(<ArticlesListing isFetching={isFetching} articles={articles} />);
  expect(wrapper).toMatchSnapshot();
});

it('it should render article item component', () => {
  const wrapper = shallow(<ArticleItem article={article} />);
  expect(wrapper.length).toBe(1);
});
it('article item component should match snapshot', () => {
  const wrapper = shallow(<ArticleItem article={article} />);
  expect(wrapper.length).toBe(1);
});
