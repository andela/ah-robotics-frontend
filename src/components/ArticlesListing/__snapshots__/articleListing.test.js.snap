// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`article listing component should match snapshot 1`] = `
<Segment
  basic={true}
  loading={false}
>
  <Grid>
    <GridColumn
      width={12}
    >
      <ItemGroup
        divided={true}
      />
    </GridColumn>
    <GridColumn
      width={4}
    />
  </Grid>
</Segment>
`;
